doc/specs/#6900 - Actions Page/spec.md

```diff
@@ -0,0 +1,130 @@
+---
+author: Kayla Cinnamon - cinnamon-msft
+created on: 2021-03-04
+last updated: 2021-03-09
+issue id: 6900
+---
+
+# Actions Page
+
+## Abstract
+
+We need to represent actions inside the settings UI. This spec goes through the possible use cases and reasoning for including specific features for actions inside the settings UI.
+
+## Background
+
+### Inspiration
+
+It would be ideal if we could get the settings UI to have parity with the JSON file. This will take some design work if we want every feature possible in relation to actions. There is also the option of not having parity with the JSON file in order to present a simpler UX.
+
+### User Stories
+
+All of these features are possible with the JSON file. This spec will go into discussion of which (possibly all) of these user stories need to be handled by the settings UI.
+
+- Add key bindings to an action that does not already have keys assigned
+- Edit key bindings for an action
+- Remove key bindings from an action
+- Add multiple key bindings for the same action
+- Create an iterable action
+- Create a nested action
+- Choose which actions appear inside the command palette
+- See all possible actions, regardless of keys
+
+Commands with properties:
+- sendInput has "input"
+- closeOtherTabs has "index"
+- closeTabsAfter has "index"
+- renameTab has "title"*
+- setTabColor has "color"*
+- newWindow has "commandline", "startingDirectory", "tabTitle", "index", "profile"
+- splitPane has "split", "commandline", "startingDirectory", "tabTitle", "index", "profile", "splitMode", "size"
+- copy has "singleLine", "copyFormatting"
+- scrollUp has "rowsToScroll"
+- scrollDown has "rowsToScroll"
+- setColorScheme has "colorScheme"
+
+Majority of these commands listed above are intended for the command palette, so they wouldn't make much sense with keys assigned to them anyway.
+
+### Future Considerations
+
+One day we'll have actions that can be invoked by items in the dropdown menu. This setting will have to live somewhere. Also, once we get a status bar, people may want to invoke actions from there.

```

Comment:
Might want to add here that we're specifically _not_ addressing customizing args as a part of this design. Action args is a big part of the actions, so I think it's good to come out strong and say "we have a plan, but we can't do it quite yet, and here's why".

Throw in that mockup with the content dialog. Then link to the upstream MUX bug tracking content dialogs not working in xaml islands and say something like "once the upstream bug is resolved, this is the UI we plan on".